### PR TITLE
samples(pubsub): add sample for eos subscription with ack results

### PR DESCRIPTION
**Pull request by NivedhaSenthil**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

**Comment by NivedhaSenthil** [comment]
To be merged after https://github.com/googleapis/google-cloud-ruby/pull/18702 and https://github.com/googleapis/google-cloud-ruby/pull/18824

**Review by dazuma — CHANGES_REQUESTED** [review]

File: google-cloud-pubsub/samples/pubsub_subscriber_exactly_once.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    PubsubSubscriberExactlyOnce.new.subscriber_exactly_once project_id: project_id,
+                                                            topic_id: topic_id,
+                                                            subscription_id: subscription_id
+  end
+end
+
+if $PROGRAM_NAME == __FILE__
+  PubsubSubscriberExactlyOnce.run
+end
+# [END pubsub_subscriber_exactly_once]
```
[the remark this answers is not in the document]
Comment (NivedhaSenthil) [reply google-cloud-pubsub/samples/pubsub_subscriber_exactly_once.rb:54]: I tried following the samples guide that we created recently.. https://github.com/GoogleCloudPlatform/samples-style-guide/pull/33/files where we wanted to have the sample being easily runnable with very minimal changes.

[the remark this answers is not in the document]
Comment (dazuma) [reply google-cloud-pubsub/samples/pubsub_subscriber_exactly_once.rb:54]: Oh geez. That style guide needs to be revisited. Having `if $PROGRAM_NAME == __FILE__` boilerplate visible in a sample is just ridiculous.

**Review by dazuma — APPROVED** [review]
LGTM

**Pull request merged**